mesosutil/mesosprotoutil.go

```diff
@@ -36,27 +36,30 @@ func FilterResources(resources []*mesos.Resource, filter func(*mesos.Resource) b
 	return result
 }
 
-func NewScalarResource(name string, val float64) *mesos.Resource {
+func NewScalarResource(name string, val float64, role *string) *mesos.Resource {
 	return &mesos.Resource{
 		Name:   proto.String(name),
 		Type:   mesos.Value_SCALAR.Enum(),
 		Scalar: &mesos.Value_Scalar{Value: proto.Float64(val)},
+		Role:   role,
 	}
 }
 
-func NewRangesResource(name string, ranges []*mesos.Value_Range) *mesos.Resource {
+func NewRangesResource(name string, ranges []*mesos.Value_Range, role *string) *mesos.Resource {
 	return &mesos.Resource{
 		Name:   proto.String(name),
 		Type:   mesos.Value_RANGES.Enum(),
 		Ranges: &mesos.Value_Ranges{Range: ranges},
+		Role:   role,
 	}
 }
 
-func NewSetResource(name string, items []string) *mesos.Resource {
+func NewSetResource(name string, items []string, role *string) *mesos.Resource {
```

Comment:
These three functions got polluted with this argument which isn't always required. It's better to create a new function which takes a slice of resources and sets the role on them.
```go
// A ResourceOpt is a functional option type for mesos.Resources.
type ResourceOpt func(*mesos.Resource)

// NewResource returns a new *mesos.Resource with the given
// ResourceOpts applied to it.
func NewResource(opts ...ResourceOpt) *mesos.Resource {
    var r mesos.Resource
    for _, opt := range opts {
        opt(&r)
    }
    return &r
}

// Role returns a ResourceOpt that sets the Role of a
// *mesos.Resource to the given value.
func Role(s string) ResourceOpt {
    return func(r *mesos.Resource) {
        r.Role = s
    }
}
```